### PR TITLE
return zeta(2*n) and zeta(-n) as expression

### LINKED ISSUE #14177: Some expressions not being evaluated at arbitrary positive integers
```
>>> n = symbols('n', positive=True, integer=True)
>>> zeta(2*n)
zeta(2*n)
>>> zeta(-n)
zeta(-n)
>>> integrate((hermite(n, x))**(2)*exp(-x**2), (x, -oo, oo))
Integral(exp(-x**2)*hermite(n, x)**2, (x, -oo, oo))
```
The values of `zeta(2*n)` and `zeta(-n)` are given in the SymPy docs and the integral (which is related to the orthogonality of the Hermite polynomials) is 
``` 
 n      
2 ⋅√π⋅n!
```

### PR DESCRIPTION
**Pull request by vishalg2235**
### Returns zeta(2*n) and zeta(-n) as expression involving Bernoulli and factorial.
Fixes #14177 .
@jksuom  please review!

### CODE REVIEW

**Comment by jksuom** [comment]
Is it possible to use the same piece of code for both `n.is_Integer` and `n.is_integer` cases?

**Comment by vishalg2235** [comment]
No @jksuom because `z.is_Number` is False. 
```
>>> n = symbols('n', positive=True, integer=True)
>>> z = 2*n
>>> z.is_Number
False
```

**Comment by vishalg2235** [comment]
@jksuom can you please have look at the error, It says "permutation error" !!

**Comment by jksuom** [comment]
It is not necessary to check `z.is_Number` as a prerequisite for checking `z.is_Integer`. I would replace `elif z.is_Integer:` with `if z.is_integer:` indented to the same level as `if z.is_Number:`.

File: sympy/functions/special/zeta_functions.py [hunk 1]
```diff
@@ -476,6 +478,14 @@ def eval(cls, z, a_=None):
                         return zeta + harmonic(abs(a), z)
                     else:
                         return zeta - harmonic(a - 1, z)
+                elif len(sym) == 1:
+                    n = sym.pop()
+                    if n.is_integer and n.is_positive:
```
Comment (jksuom) [inline sympy/functions/special/zeta_functions.py]: Could this be `if z.is_positive and z.is_even:`?

**Comment by jksuom** [comment]
Is it necessary to consider different cases for `len(sym)`?

**Comment by vishalg2235** [comment]
one case is when it has no symbol(z = 5) and other when it has a symbol (z = 2*n).

**Comment by jksuom** [comment]
Would `z.is_even and z.is_positive` depend on whether `z` contains a symbol or not?

**Comment by vishalg2235** [comment]
@jksuom I got what you are telling. Now there is single common code for `z = 5` (Integer) and `z = 2*n` (integer) as in both case `z.is_integer = True` and please review

File: sympy/functions/special/zeta_functions.py [hunk 2]
```diff
@@ -462,20 +462,27 @@ def eval(cls, z, a_=None):
...
-
-                    if a.is_negative:
-                        return zeta + harmonic(abs(a), z)
-                    else:
-                        return zeta - harmonic(a - 1, z)
+        if z.is_integer:
+            if a.is_Integer:
+                if z.is_negative:
+                    zeta = (-1)**z * bernoulli(-z + 1)/(-z + 1)
+                elif z.is_even:
```
Comment (jksuom) [inline sympy/functions/special/zeta_functions.py]: I think we should add `and z.is_positive` to take care of those cases where the sign of  `z` cannot be determined. (It contains a symbol.) Then `None` should be returned.

**Comment by vishalg2235** [comment]
@jksuom please review!

**Comment by jksuom** [comment]
This looks good. But would it be possible to drop the `harmonic` part from the unit tests now that #14186 has been merged?

**Comment by Roland-Sands** [comment]
For `zeta(2*n)` I think it is better to have a power of -1 rather than take the absolute value of `B_2n`

**Comment by jksuom** [comment]
I think so too. That is more informative.

**Comment by Roland-Sands** [comment]
The line that I had in mind is this one:
`assert zeta(2*n) == 2**(2*n - 1)*pi**(2*n)*Abs(bernoulli(2*n))/factorial(2*n)`

**Comment by Roland-Sands** [comment]
@vishalg2235 Thumbs up!!! Excellent!!

If the functional equation for the zeta function was implemented (perhaps it is there and I missed it) then each of `zeta(2*n)` and `zeta(-n)` implies the other

**Comment by vishalg2235** [comment]
@jksuom please review.

**Comment by vishalg2235** [comment]
Ping @jksuom !

**Comment by jksuom** [comment]
Thanks, this looks good. (This will close the issue that also involves Hermite functions, but I think that more detailed plans would be needed for that anyway.)

**Pull request merged**